### PR TITLE
[v3.0.0] Implement context-aware guidance system for all commands

### LINKED ISSUE #206: [v3.0.0] Implement context-aware guidance system for all commands
## Overview

Implement context-aware guidance system that provides intelligent "Next Steps" suggestions after each command execution.

## Related

- Concept: docs/v3/concept.md - Phase 5: Context-Aware Guidance
- Foundation: #194 (Guidance system foundation)

## Design Principle

After each command execution, clearly present context-appropriate next actions based on current state.

## Guidance Format

```
[Command execution result]

Next Steps
──────────
→ [Highest priority action]
  [Specific command]

→ [Alternative action]
  [Alternative command]
```

### 1. After New Task Generation (with TODO)
```
→ Start working on TODO tasks
  reviewtask show       # See next recommended task

→ View all tasks
  reviewtask status
```

### 2. After New Task Generation (with PENDING)
```
→ Start working on TODO tasks first
  reviewtask show

! You have PENDING tasks requiring decision
  Review them after completing TODO tasks
```

### 3. After Task Completion (next TODO exists)
```
→ Continue with next task
  reviewtask show

→ Start immediately
  reviewtask start <next-task-id>
```

### 4. TODO Complete, PENDING Exists
```
! All TODO tasks completed

You have PENDING tasks requiring your decision:

PENDING Tasks
─────────────
abc123  Refactor authentication module
def456  Consider using dependency injection

Next Steps
──────────
→ Review PENDING tasks
  reviewtask show abc123

→ Decide: start or cancel
  reviewtask start abc123   # Start working
  reviewtask cancel abc123  # Skip this task
```

### 5. All Complete (TODO + PENDING)
```
✓ All tasks completed

→ Push your changes
  git push

→ Check for new reviews
  reviewtask

Your PR is ready for final review
```

### 1. Context Detection

Implement detector to analyze current state:
- Number of tasks by status (TODO, DOING, PENDING, DONE)
- Presence of unresolved comments
- Current task state
- Git staging state (for commit suggestions)

### 2. Guidance Rules

Define rules for each context:
```go
type GuidanceContext struct {
    TodoCount    int
    PendingCount int
    DoingCount   int
    DoneCount    int
    UnresolvedComments int
    CurrentTaskID string
    // ...
}

func (g *Guidance) Suggest(ctx GuidanceContext) []Action
```

### 3. Integration Points

Add guidance to:
- fetch/analyze (after task generation)
- status (based on current state)
- done (after task completion)
- start (after starting task)
- hold (after holding task)
- cancel (after cancellation)
- show (when viewing task details)

## Implementation Files

```
internal/
  guidance/
    guidance.go       # Core guidance system
    detector.go       # Context detection logic
    formatter.go      # Format guidance output
    rules.go          # Guidance rule definitions
```

## Example Integration

```go
// In cmd/done.go
func runDone(cmd *cobra.Command, args []string) error {
    // ... task completion logic ...
    
    // Get guidance
    ctx := guidance.DetectContext(storageManager)
    suggestions := guidanceSystem.Suggest(ctx)
    
    // Display
    fmt.Println()
    guidance.Display(suggestions)
    
    return nil
}
```

## Success Criteria

- All commands show context-appropriate guidance
- Guidance accurately reflects current state
- Suggestions are actionable (valid commands)
- Guidance prioritizes most important action first
- Warning indicators for PENDING tasks
- Clear completion messages when all done
- Tests covering guidance logic
- Documentation includes guidance examples

### PR DESCRIPTION
**Pull request by biwakonbu**
Closes #206

## Overview

Implements context-aware guidance system that provides intelligent "Next Steps" suggestions after each command execution.

## Implementation

- [x] Context detection logic (detector.go)
- [x] Guidance rules system (rules.go)
- [x] Guidance formatter (formatter.go)
- [x] Integration into commands (done, status, start)
- [x] Comprehensive tests (96.2% coverage)

### Core Implementation

1. **detector.go**: Analyzes current state to build guidance context
   - Counts tasks by status (TODO, DOING, DONE, PENDING, HOLD)
   - Detects completion states and next recommended tasks
   - Language configuration support

2. **rules.go**: Priority-based rule system for guidance generation
   - UnresolvedCommentsRule (priority 100)
   - AllCompleteRule (priority 90)
   - PendingOnlyRule (priority 80)
   - ContinueTasksRule (priority 70)
   - StartTasksRule (priority 60)
   - DefaultRule (fallback, priority 0)

3. **formatter.go**: Multiple output formats
   - Standard format with section dividers
   - Compact format for concise output
   - With-context format including task summary

### Command Integration

- **done**: Shows next steps after task completion
- **status**: Displays guidance based on current state
- **start**: Provides guidance after starting work

### Unit Tests (96.2% Coverage)

- **detector_test.go**: 2 test functions, 7 scenarios
  - Empty state, todo only, all done, mixed statuses
  - Pending tasks, language configuration
  
- **formatter_test.go**: 6 test functions
  - Basic, compact, and context-aware formatting
  - Edge cases: empty guidance, long descriptions

- **rules_test.go**: 8 test functions
  - All rule types and priority matching
  - Rule set customization

### After Task Completion
```
Next Steps
──────────
→ Continue with next task
  reviewtask show

→ Start immediately
  reviewtask start abc123
```

### When TODO Complete, PENDING Exists
```
Next Steps
──────────
! All TODO tasks completed

You have 3 PENDING tasks requiring your decision:

→ Review PENDING tasks
  reviewtask show <task-id>

→ Decide: start or cancel
  reviewtask start <task-id>
```

### All Complete
```
Next Steps
──────────
✓ All tasks completed

→ Push your changes
  git push

→ Check for new reviews
  reviewtask
```

## Progress

✅ Implementation complete
✅ All tests passing
✅ 96.2% test coverage

## Summary by CodeRabbit

* **New Features**
  * Context-aware guidance now appears after starting tasks, viewing AI status, and finishing work — includes "Next Steps", task summary, and suggested commands.

* **UX**
  * Integrated workflow: single command accepts an optional PR number and auto-runs analysis.
  * "done" replaces the prior "complete" command in user-facing flows.

* **Refactor**
  * Guidance generation moved to a rule-driven system while preserving legacy next-step behavior.

* **Documentation**
  * Command docs, examples, and help updated for the integrated workflow and renamed commands.

* **Tests**
  * Added comprehensive guidance tests; removed several command-specific tests.

### CODE REVIEW

File: internal/guidance/detector.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		case StatusDoing:
+			ctx.DoingCount++
+		case StatusDone:
+			ctx.DoneCount++
+		case StatusPending:
+			ctx.PendingCount++
+		case StatusHold:
+			ctx.HoldCount++
+		}
+	}
```
[the remark this answers is not in the document]
Comment (biwakonbu) [reply internal/guidance/detector.go:57]: **Task Cancelled** (Priority: HIGH)

**Original Feedback:**
> Add StatusCancel case to task counting switch in detector.go to treat cancelled tasks as complete

**Cancellation reason:**
Testing cancel during dogfooding

[the remark this answers is not in the document]
Comment (biwakonbu) [reply internal/guidance/detector.go:57]: **Task Cancelled** (Priority: HIGH)

**Original Feedback:**
> Add StatusCancel case to task counting switch in detector.go to properly handle cancelled tasks

**Cancellation reason:**
Duplicate of task e1c65566-0e2c-4c5a-86be-663a068af5bf which addresses the same StatusCancel case issue

ℹ️ This comment has 1 other task(s) still active

File: docs/v3/README.md [hunk 2]
```diff
@@ -16,18 +16,26 @@ Main v3.0.0 design document covering:
...
+### Phase 4: Automation (Week 9-11) ✅ COMPLETED
+- [#195](https://github.com/biwakonbu/reviewtask/issues/195) - ✅ Implement done command automation routine
+
+### Phase 5: Guidance Enhancement ✅ COMPLETED
+- [#206](https://github.com/biwakonbu/reviewtask/issues/206) - ✅ Implement context-aware guidance system for all commands
+
+### Remaining for v3.0.0 Final Release 🚧
+- [#215](https://github.com/biwakonbu/reviewtask/issues/215) - 🚧 Implement integrated reviewtask command (fetch + analyze)
+- [#216](https://github.com/biwakonbu/reviewtask/issues/216) - 🚧 Remove or deprecate analyze command flags
+- [#217](https://github.com/biwakonbu/reviewtask/issues/217) - 🚧 Deprecate complete command in favor of done
```
[the remark this answers is not in the document]
Comment (biwakonbu) [reply docs/v3/README.md]: **Task Cancelled** (Priority: MEDIUM)

**Original Feedback:**
> Remove 'Remaining for v3.0.0 Final Release 🚧' subsection from docs/v3/README.md to comply with documentation guidelines

**Cancellation reason:**
Duplicate of task 1d45de66 which was already completed in commit 324b20b

File: internal/guidance/detector.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	// Find next suggested task (highest priority TODO task)
+	if ctx.TodoCount > 0 {
+		for _, task := range allTasks {
+			if task.Status == StatusTodo {
+				ctx.NextTaskID = task.ID
+				ctx.NextTaskDesc = task.Description
+				break
+			}
+		}
+	}
```
[the remark this answers is not in the document]
Comment (biwakonbu) [reply internal/guidance/detector.go:80]: **Task Cancelled** (Priority: MEDIUM)

**Original Feedback:**
> Implement priority-based task selection in next task suggestion logic

**Cancellation reason:**
Duplicate of task 80bef35e which was already completed in commit 56e6bd1

File: docs/user-guide/commands.md [hunk 4]
```diff
@@ -238,9 +238,11 @@ reviewtask verify task-uuid-here
 - `feature-task`: Tasks containing "feature" or "implement"
 - `general-task`: All other tasks
 
-#### `reviewtask complete <task-id> [options]`
+#### `reviewtask complete <task-id> [options]` **[DEPRECATED]**
 
-Complete task with automatic verification.
+> **⚠️ Deprecation Notice**: This command will be removed in v3.0.0. Use `reviewtask done` instead for full workflow automation.
+
+Complete task with automatic verification (limited functionality).
```
[the remark this answers is not in the document]
Comment (biwakonbu) [reply docs/user-guide/commands.md]: **Task Cancelled** (Priority: MEDIUM)

**Original Feedback:**
> Update deprecation notice wording for 'complete' command to clarify it's deprecated in v3.0.0 but remains available for backward compatibility

**Cancellation reason:**
Complete command was removed entirely in v3.0.0 (commit c7680ba), not just deprecated, so no deprecation notice exists to update

ℹ️ This comment has 1 other task(s) still active

[the remark this answers is not in the document]
Comment (biwakonbu) [reply docs/user-guide/commands.md]: **Task Cancelled** (Priority: LOW)

**Original Feedback:**
> Update deprecation notice wording for 'complete' command to clarify it is deprecated in v3.0.0 but retained for backward compatibility

**Cancellation reason:**
Duplicate of task 253d07a5 - both refer to updating complete command deprecation notice which no longer exists after complete command removal in v3.0.0

File: internal/guidance/detector.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (biwakonbu) [reply internal/guidance/detector.go:57]: **Task Cancelled** (Priority: HIGH)

**Original Feedback:**
> Add StatusCancel case to task counting switch statement in detector.go

**Cancellation reason:**
Already implemented in commit 2571ea7: StatusCancel case added to task counting switch in detector.go

File: docs/v3/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (biwakonbu) [reply docs/v3/README.md]: **Task Cancelled** (Priority: MEDIUM)

**Original Feedback:**
> Remove 'Remaining for v3.0.0 Final Release 🚧' subsection from docs/v3/README.md to comply with documentation guidelines

**Cancellation reason:**
Already completed in commit 324b20b: The 'Remaining for v3.0.0 Final Release' subsection was removed from docs/v3/README.md

File: docs/user-guide/commands.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (biwakonbu) [reply docs/user-guide/commands.md]: **Task Cancelled** (Priority: MEDIUM)

**Original Feedback:**
> Remove the 'reviewtask complete' command section from docs/user-guide/commands.md since the command was fully removed in v3.0.0

**Cancellation reason:**
Already completed in commit 2b374f4: All references to complete command including the documentation section were removed from docs/user-guide/commands.md
